tests/test_dimension.py

```diff
@@ -1102,19 +1099,40 @@ def test_no_fusion_convoluted(self):
                 Eq(g, f + 1)]
 
         op = Operator(eqns)
+        trees = retrieve_iteration_tree(op)
+        assert(len(trees) == 4)
+
+        assert(len(trees[0]) == 6)
+        exprs = FindNodes(Expression).visit(trees[0])
```

Comment:
since neither `exprs` nor `trees_in` are accessed more than once, you don't actually need these variables. Just `assert len(FindNodes(....))`